test/e2e/networkpolicy_test.go

```diff
@@ -0,0 +1,144 @@
+
+	npDenyAll := "test-networkpolicy-deny-all"
+	denyAllPolicy, err := data.createNetworkPolicyDenyAll(npDenyAll)
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(denyAllPolicy); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	workerNode := workerNodeName(1)
+
+	podName0 := randName("test-pod-networkpolicy-")
+	if err := data.createBusyboxPodOnNode(podName0, workerNode); err != nil {
+		t.Fatalf("Error when creating busybox test Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, podName0)
+	if _, err := data.podWaitForIP(defaultTimeout, podName0); err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", podName0, err)
+	}
+
+	podName1 := randName("test-pod-networkpolicy-")
+	if err := data.createBusyboxPodOnNode(podName1, workerNode); err != nil {
+		t.Fatalf("Error when creating busybox test Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, podName1)
+	podIP1, err := data.podWaitForIP(defaultTimeout, podName1)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", podName1, err)
+	}
+
+	nginxPodName := randName("test-pod-nginx-")
+	if err = data.createNginxPodOnNode(nginxPodName, workerNode); err != nil {
+		t.Fatalf("Error when creating nginx pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, nginxPodName)
+	if _, err := data.podWaitForIP(defaultTimeout, nginxPodName); err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", nginxPodName, err)
+	}
+
+	svcName := randName("test-svc-nginx")
+	if err = data.createNginxService(svcName); err != nil {
+		t.Fatalf("Error when creating nginx service: %v", err)
+	}
+	defer func() {
+		if err = data.deleteService(svcName); err != nil {
+			t.Fatalf("delete nginx service error: %v", err)
+		}
+	}()
+
+	// Two pods cannot wget to service.
+	if err = data.runWgetCommandFromTestPod(podName0, svcName); err == nil {
+		t.Fatalf("Pod %s should not be able to connect Service %s, but was able to connect", podName0, svcName)
+	}
+	if err = data.runWgetCommandFromTestPod(podName1, svcName); err == nil {
+		t.Fatalf("Pod %s should not be able to connect Service %s, but was able to connect", podName1, svcName)
+	}
+
+	npIPblockExcept := "test-networkpolicy-ipblock-except"
+	IPBlockExceptPolicy, err := data.createNetworkPolicyIPBlockWithExcept(npIPblockExcept, podIP1)
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(IPBlockExceptPolicy); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	// Now pod1 cannot wget to service.
+	if err = data.runWgetCommandFromTestPod(podName1, svcName); err == nil {
+		t.Fatalf("Pod %s should not be able to connect Service %s, but was able to connect", podName1, svcName)
+	}
+	// Now pod0 can still wget to service.
+	if err = data.runWgetCommandFromTestPod(podName0, svcName); err != nil {
```

Comment:
pod0 didn't wget to service before so it's not `can still`. I think you should test pod0 can wget to service first because that would wait for the policy being enforced implicitly, otherwise we don't know pod1 can't wget is because the policy is not enforced or the policy is enforced.

Reply:
It makes sense. Updated.